src/Raven.Server/Documents/Handlers/DocumentHandler.cs

```diff
@@ -425,17 +425,17 @@ private static void ThrowFailedToUpdateHash()
             {
                 var id = GetQueryStringValueAndAssertIfSingleAndNotEmpty("id");
 
-                var doc = await context.ReadForDiskAsync(RequestBodyStream(), id);
+                var doc = context.ReadForDiskAsync(RequestBodyStream(), id).ConfigureAwait(false);
 
-                if (id.EndsWith("/"))
+                if (id[id.Length-1] == '/')
                 {
                     var (_, clusterId) = await ServerStore.GenerateClusterIdentityAsync(id, Database.Name);
                     id = clusterId;
                 }
 
                 var changeVector = context.GetLazyString(GetStringQueryString("If-Match", false));
 
-                var cmd = new MergedPutCommand(doc, id, changeVector, Database);
+                var cmd = new MergedPutCommand(await doc, id, changeVector, Database);
```

Comment:
My hypothesis is that ReadForDiskAsync is requiring to do some setup that is forcing anyways the use of an async thread (so it will block anyways), therefore all the extra execution that happens in between becomes essentially free from the latency point of view.